improver_tests/nowcasting/optical_flow/test_OpticalFlow.py

```diff
@@ -911,16 +913,14 @@ def test_error_nonmatching_inputs(self):
         with self.assertRaisesRegex(ValueError, msg):
             self.plugin.process(self.cube1, self.cube2)
 
-    @ManageWarnings(record=True)
-    def test_warning_nonprecip_inputs(self, warning_list=None):
+    def test_warning_nonprecip_inputs(self):
         """Test code raises a warning if input cubes have
         non-rain variable names"""
         self.cube1.rename("snowfall_rate")
         self.cube2.rename("snowfall_rate")
-        _, _ = self.plugin.process(self.cube1, self.cube2, boxsize=3)
         warning_msg = "Input data are of non-precipitation type"
-        self.assertTrue(any(item.category == UserWarning for item in warning_list))
-        self.assertTrue(any(warning_msg in str(item) for item in warning_list))
+        with pytest.warns(UserWarning, match=warning_msg):
+            _, _ = self.plugin.process(self.cube1, self.cube2, boxsize=3)
```

Comment:
You've done this exactly as it was before, but it should be simplified to:
```suggestion
            self.plugin.process(self.cube1, self.cube2, boxsize=3)
```